### PR TITLE
refactor(admin-panel, payments): Upgrade to react-router v6

### PR DESCRIPTION
**Pull request by LZoog**
## Because

- This dependency couldn't be automatically updated by dependabot due to breaking changes which prevents us from upgrading various other packages

## This pull request

- Upgrades the admin panel to react-router v6
- Upgrades the payments-server to react-router v6
- Contains very small amounts of payments-server cleanup on touched files

## Issue that this pull request solves

Closes: #11188

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.

---

Would probably like to get @jodyheavener's eyes on this as well as someone familiar with SubPlat flows to poke around and ensure nothing is broken.

~I'm opening this as a draft because something funky is going on. I tested the subplat flow after my changes and it worked all fine locally, but I had some dependency conflicts after rebasing and now I'm getting a general app error. Need to investigate. *Edit: yep def have a dependency conflict problem.*~ Fixed.

Also, when working on getting payments-server tests running, came across and filed #11460. If you see a warning in payments after these test fixes, that's why.

### CODE REVIEW

**Comment by jodyheavener** [comment]
Some fine tuning is probably possible here, but I was able to resolve the dependency issue, fix the remaining tests, and get Storybook working again. I tested the flows locally and it _seems_ like it's all working. ✨

**Comment by LZoog** [comment]
Thanks @jodyheavener for circling back here :) I checked it out and it LGTM.

It's probably good if someone from SubPlat reviews this.

File: packages/fxa-payments-server/package.json [hunk 1]
```diff
@@ -45,7 +45,7 @@
     "@babel/core": "^7.15.0",
     "@babel/register": "^7.16.5",
     "@fluent/bundle": "^0.17.1",
-    "@fluent/langneg": "^0.6.1",
```
Comment (chenba) [inline packages/fxa-payments-server/package.json:48]: oh?

File: packages/fxa-payments-server/src/routes/Product/IapRoadblock/index.tsx [hunk 2]
```diff
@@ -42,6 +42,7 @@ export const IapRoadblock = ({
   subscription,
 }: IapRoadblockProps) => {
   const { l10n } = useLocalization();
+  // const navigate = useNavigate();
```
Comment (chenba) [inline packages/fxa-payments-server/src/routes/Product/IapRoadblock/index.tsx:45]: Let's delete it.

File: packages/fxa-payments-server/src/routes/Product/IapRoadblock/index.tsx [hunk 3]
```diff
@@ -50,6 +51,8 @@ export const IapRoadblock = ({
   const title = (
     <SubscriptionTitle screenType="iapsubscribed" subtitle={subtitle} />
   );
+
+  // const manageSubscription: VoidFunction = () => navigate(appStoreLink);
```
Comment (chenba) [inline packages/fxa-payments-server/src/routes/Product/IapRoadblock/index.tsx:55]: Delete?

**Review by chenba — APPROVED** [review]
Thanks for making these changes!  Looks there's an accidental(?) module downgrade.

**Comment by LZoog** [comment]
Thanks @chenba! Oops, I'm not sure how those console logs were left around, I'll remove those and look at the dependency downgrade.

**Pull request merged**